### PR TITLE
Introduce libs/viewer and serializable view settings.

### PR DESCRIPTION
**Pull request by prideout**
NOTE: individual commits might be easier to review than the consolidated PR.

Adds a new lib that holds SimpleViewer and a new `Settings` structure that includes a JSON reader / writer. This will be used for automated testing and for client / server communication.

The `viewer::Settings` struct is closely associated with the `filament::View` API; when updating the latter we will often need to update the former, as well as some serialization code. This increases the maintenance burden and I think we should consider using a parser library like libclang or a macro-based reflection utility.

This PR also migrates SimpleViewer into libs/viewer and un-inlines its implementation. It does not belong in gltfio because it has an imgui dependency.

### CODE REVIEW

**Review by romainguy — APPROVED** [review]
👍

File: libs/viewer/src/Settings.cpp [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+#include <viewer/Settings.h>
+
+#include <utils/Log.h>
+
+#include <sstream>
+#include <string>
+
+#include <assert.h>
+
+// NOTE: We are including cgltf only to get JSMN. We should consider adding jsmn to third_party.
```
Comment (bejado) [inline libs/viewer/src/Settings.cpp]: Is this comment still needed / can we include the jsmn headers directly now?

**Review by bejado — APPROVED** [review]
Very excited for this!

File: libs/viewer/src/Settings.cpp (hunk 1 above)
Comment (prideout) [reply libs/viewer/src/Settings.cpp]: I removed this comment in a subsequent CL.

Comment (bejado) [reply libs/viewer/src/Settings.cpp]: whoops, missed that

**Pull request merged**